backend/src/models/services/entry/entry.ts

```diff
@@ -79,135 +123,307 @@ export async function getEntryConversation(entryId: string) {
+  const llmResponse = await newConversation.getChatContent(
+    configId,
+    entryAnalysis.id,
+    messageData.messages[0].message_content
+  );
+  
+  if (llmResponse) {
+    // Messages defined because messageData.messages defined
+    newConversation.messages![0].llm_response = llmResponse;
   }
+  await newConversation.save();
+  await entry.save();
 
-  const cdGpt = new CdGpt(process.env.OPENAI_API_KEY, config.model.analysis);
+  return newConversation;
+}
 
-  cdGpt.seedAnalysisMessages();
-  cdGpt.addUserMessage({ analysis: content });
+/**
+ * Updates EntryConversation associated with chatId with messages in messageData
+ * and generates LLM response to them.
+ * 
+ * @param chatId id of EntryConversation
+ * @param configId id of Config for LLM
+ * @param messageData messages to update EntryConversation with
+ * @returns Updated EntryConversation
+ */
+export async function updateEntryConversation(
+  chatId: string,
+  configId: string,
+  messageData: UpdateChatRequestBody
+) {
+  const { conversation, analysis } = await _verifyEntryConversation(chatId);
 
-  //TODO: replace type with better fitting one
-  const analysisCompletion: any = await cdGpt.getAnalysisCompletion();
+  return await _populateChatContent(configId, analysis, messageData, conversation);
+}
 
-  if (analysisCompletion.error) {
-    throw new Error(analysisCompletion.error.message);
+/**
+ * Private function for checking if Entry and EntryAnalysis associated with
+ * entryId exist.
+ * 
+ * @param entryId id associated with Entry and EntryAnalysis to check
+ * @returns Entry and EntryAnalysis associated with entryId
+ */
+async function _verifyEntry(entryId: string) {
+  const entry = await getEntryById(entryId);
+  if (!entry) {
+    throw new Error('Entry not found.');
   }
+  const entryAnalysis = await getEntryAnalysisById(entryId);
+  if (!entryAnalysis) {
+    throw new Error('Entry analysis not found.');
+  }
+  return { entry, entryAnalysis };
+}
 
-  const response = JSON.parse(analysisCompletion.choices[0].message.content);
-
-  const { reframed_thought: reframing, distortion_analysis: analysis, impact_assessment: impact, affirmation, is_healthy: isHealthy } = response;
-
-  if (!isHealthy) {
-    if (!analysis || !impact || !reframing) {
-      throw new Error('Analysis content is not available.');
+/**
+ * Private function for updating Entry and EntryAnalysis with LLM content.
+ * 
+ * @param updatedEntry Entry to update
+ * @param oldAnalysis EntryAnalysis to update with LLM content
+ * @param configId id of Config for LLM
+ * @returns Entry and EntryAnalysis updated with LLM content, and error message on error
+ */
+async function _updateEntry(
+  updatedEntry: HydratedDocument<EntryType>,
+  // Can't use HydratedDocument<EntryAnalysisType> because it doesn't have model methods
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
Yes it's related to the CdGpt refactor #173.

So if I wanted to avoid the explicit any type, I'd do something like this for the function header:
```
async function _updateEntry(
  updatedEntry: HydratedDocument<EntryType>,
  oldAnalysis: HydratedDocument<EntryAnalysisType, EntryAnalysisMethods>,
  configId: string
)
```

This would require exporting `EntryAnalysisMethods` in entryAnalysis.ts and importing it here, which feels off because it clutters the namespace with something that's gonna be used just here, and that we're planning on changing anyways. So I felt it was appropriate to use `any` and change this with #173